service/controller/clusterapi/v21/resource/operatorversions/delete.go

```diff
@@ -2,15 +2,14 @@ package operatorversions
 
 import (
 	"context"
-
-	"github.com/giantswarm/microerror"
 )
 
+// EnsureDeleted is not putting the operator versions into the controller
+// context because we do not want to fetch the version information on delete
+// events. This is to reduce eventual friction. Cluster deletion should not be
+// affected only because some releases are missing or broken when fetching them
+// from cluster-service. Other resources must not rely on operator version
+// information on delete events.
```

Comment:
I just had a case where releases where screwed and some test cluster was hanging there and couldn't be deleted. This should help.